processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessPipeline.java

```diff
@@ -124,6 +125,19 @@ Completion process(TaskRequest request, DecatonTask<T> task) throws InterruptedE
                 new ProcessingContextImpl<>(scope.subscriptionId(), request, task, processors, retryProcessor,
                                             scope.props());
 
+        if (task.metadata().timestampMillis() > 0) {
```

Comment:
Maybe we better to filter-out `currentTimeMillis() - timestampMillis() < 0` cases to avoid counting obviously broken timestamp value? (and likewise for the below `tasksScheduledDelay` case?)

Reply:
As https://github.com/line/decaton/pull/218#discussion_r1403968994 said, indeed the filter-out is necessary because we can freely modify the metadata timestamp. Thank you!